pkg/kademlia/kademlia_test.go

```diff
@@ -189,7 +201,10 @@ func testNode(t *testing.T, bn []pb.Node) (*Kademlia, *grpc.Server) {
 	id := dht.NodeID(fid.ID)
 	assert.NoError(t, err)
 	// new kademlia
-	k, err := NewKademlia(id, bn, lis.Addr().String(), fid, "db", kc)
+
+	dir, err := ioutil.TempDir("", "kad_test")
```

Comment:
What's cleaning up this temp directory?

Reply:
It's not about Go it's about the system. You tried to delete the whole system temp directory; which also may crash the system.

Usually you would use tempdir with cleanup like this:

```
tempdir, err := ioutil.TempDir("", "kad_test")
if err != nil {
    t.Fatal(err)
}
defer func() {
    err := os.RemoveAll(tempdir)
    if err != nil {
        t.Fatal(err)
    }
}()
```

This is of course when you are finishing up the directory in the same func where you create the directory.